service/storage_service.cc

```diff
@@ -534,12 +531,14 @@ future<storage_service::nodes_to_notify_after_sync> storage_service::sync_raft_t
 
     std::vector<future<>> sys_ks_futures;
 
-    auto process_left_node = [&] (raft::server_id id, locator::host_id host_id, std::optional<gms::inet_address> ip) -> future<> {
+    auto process_left_node = [&] (raft::server_id id, locator::host_id host_id, std::optional<gms::inet_address> ip, bool notify) -> future<> {
         if (ip) {
             sys_ks_futures.push_back(_sys_ks.local().remove_endpoint(*ip));
 
             co_await _gossiper.force_remove_endpoint(host_id, gms::null_permit_id);
```

Comment:
It looks like we just keep the Host ID -> IP mapping as expiring for some time: `_gossiper.force_remove_endpoint` -> `gossiper.evict_from_membership` -> `g._address_map.set_expiring`. Is there a reason for this? If not, shouldn't we change the address map instead?